src/lib/notification_service/sender/smtp.py

```diff
@@ -67,12 +60,24 @@ def _get_file_mime_object(file: File) -> MIMEBase:
     def send(self) -> None:
         """Send a message via SMTP."""
         context = ssl.create_default_context()
+        smtp_secret_name = self._delivery_method.get("credentials_secret_name")
 
-        with SMTP_SSL(self._server, self._port, context=context) as server:
+        if smtp_secret_name is None:
+            raise KeyError("Credentials Secret Name is not set.")
+
+        smtp_secret = json.loads(self._secret_client.get_secret(smtp_secret_name))
```

Comment:
Maybe, it also make sense to add check if secret's content align with our expectations :) 
Because we ask end-user to create this secret